### DISCUSSION TITLE
How to set an attached texture?

### DISCUSSION THREAD

**Discussion by kirinzer** [google/filament#3923] [Q&A] [answered]
Now I found that one entity can only bind one texture as an input source. and I want to do some extra effects for one entity. So may I have some API to called and get a temp texture before rendering to the entity?

---

the current workflow like this:
A resource->upload texture->render to entity

---

anticipation:
A resource->upload texture->[some effects]->[render to temp texture]->render to entity

**Comment by romainguy** [comment, marked as answer]
I'm not sure I understand, what do you mean by "one entity can only bind one texture as an input source"? Textures are not bound to entities, they are usually bound to material instances, and you can definitely bind more than one. Also, if you need a temporary texture, just create a texture?

**Reply by kirinzer** [reply]
I could create a texture no doubt! But my confusion is how to make multi-textures to join the rendering process. Just as you said, at the same time I could create more than one material instance. I want to control them render one by one, and make the previous output as the next input.

**Reply by kirinzer** [reply]
Actually, I need to create a frameBuffer attached texture (in OpenGL backend). But I do not know which API should be chosen in Filament.

**Reply by romainguy** [reply]
You're looking for the RenderTarget API.

**Reply by kirinzer** [reply]
Thanks!! This is exactly what I want!!!

**Reply by kirinzer** [reply]
@romainguy All right, last question! After have check the RenderTarget associated API. I found that only use Filament::View object can set render target. Is there any way to set it for an entity or a material instance? Because in my case, I only create a single Filament::View object. Although there are many entities, I just want to do a post-processing effect for a specified entity.

**Reply by romainguy** [reply]
You would simply create a scene, put your entity in that scene, attach the scene to the view and render the whole thing in the render target. If your processing is cheap enough you could skip all this and just do it in a custom material as well.

**Reply by kirinzer** [reply]
I get it! It is so clear. I will try the second solution (custom material). Thanks for your patience.
